tools/codegen/pkg/bootstrapgen/generator.go

```diff
@@ -0,0 +1,47 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package bootstrapgen
+
+import (
+	"os"
+)
+
+// Generator creates a Go file that will be build inside mixer framework. The generated file contains all the
+// template specific code that mixer needs to add support for different passed in templates.
+type Generator struct {
+	OutFilePath   string
+	ImportMapping map[string]string
+}
+
+// Generate creates a Go file that will be build inside mixer framework. The generated file contains all the
+// template specific code that mixer needs to add support for different passed in templates.
+func (g *Generator) Generate(fdsFiles []string) error {
+
+	// TODO Complete the imlementation here.. Next PR.
+	// ..
+
+	f1, err := os.Create(g.OutFilePath)
+	if err != nil {
+		return err
+	}
+	defer f1.Close()
+	if _, err = f1.Write([]byte{}); err != nil {
+		_ = f1.Close()
```

Comment:
I had the same question. It seems to work fine because the defer ignores the error. Closing the file again gives already closed error, which we ignore.